code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -288,3 +317,266 @@
+	to_chat(C, "<span class='danger'>You feel a tiny prick.</span>") //Fluff
+	to_chat(X, "<span class='xenowarning'>Our stinger silently injects our victim!</span>")
+	X.playsound_local(C, 'sound/effects/spray3.ogg', 5, 0, 1)
+	C.playsound_local(C, 'sound/effects/spray3.ogg', 5, 0, 1)
+
+	succeed_activate()
+
+	GLOB.round_statistics.hunter_stings++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "hunter_stings") //Statistics
+	add_cooldown()
+
+
+// ***************************************
+// *********** Hunter's Mark
+// ***************************************
+/datum/action/xeno_action/activable/hunter_mark
+	name = "Hunter's Mark"
+	action_icon_state = "hunter_mark"
+	mechanics_text = "Psychically mark a creature you have line of sight to, allowing you to sense its direction, distance and location."
+	plasma_cost = HUNTER_STINGER_PLASMA_COST
+	keybind_signal = COMSIG_XENOABILITY_HUNTER_MARK
+	cooldown_timer = HUNTER_MARK_COOLDOWN
+	var/mob/hunter_mark_target
+
+/datum/action/xeno_action/activable/hunter_mark/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(QDELETED(A))
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/hunter/X = owner
+	var/mob/C = A
+
+	if(!istype(C))
+		to_chat(X, "<span class='xenowarning'>We cannot psychically mark this target!</span>")
+		return FALSE
+
+	if(C == X)
+		to_chat(X, "<span class='xenowarning'>Why would we target ourselves?</span>")
+		return FALSE
+
+	if(X.on_fire)
+		to_chat(X, "<span class='xenowarning'>We're too busy being on fire to mark them!</span>")
+		return FALSE
+
+	if(!X.line_of_sight(C)) //Need line of sight.
+		to_chat(X, "<span class='xenowarning'>We require line of sight to mark them!</span>")
+		return FALSE
+
+	return TRUE
+
+
+/datum/action/xeno_action/activable/hunter_mark/on_cooldown_finish()
+	var/mob/living/carbon/xenomorph/X = owner
+	to_chat(X, "<span class='xenowarning'><b>We are able to mark another with a psychic connection.</b></span>")
+	X.playsound_local(X, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	return ..()
+
+
+/datum/action/xeno_action/activable/hunter_mark/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/M = A
+
+	if(!istype(M)) //Sanity
+		fail_activate()
+
+	hunter_mark_target = M //Set our target
+
+	X.face_atom(M) //Face towards the target so we don't look silly
+
+	var/name = sanitize(M.name)
+
+	to_chat(X, "<span class='xenodanger'>We prepare to psychically mark [name] as our quarry.</span>")
+
+	if(!do_after(X, HUNTER_MARK_WINDUP, TRUE, target, BUSY_ICON_HOSTILE)) //Slight wind up
+		return fail_activate()
+
+	if(!X.line_of_sight(M)) //Need line of sight.
+		to_chat(X, "<span class='xenowarning'>We lost line of sight to the target!</span>")
+		return fail_activate()
+
+	var/datum/action/xeno_action/psychic_trace/P = locate() in X.xeno_abilities //Gotta reference the datum
+	P.trace_target = hunter_mark_target //Set Psychic Trace's target
+
```

Comment:
Instead of these locate() in abilities just add new xeno var for the tracked mob, see how xeno spits does it
Also Register signals for when the mob is qdeleting to clear references

Reply:
Done.